### PR TITLE
🌱  Use osv-scanner workflow file instead of local workflow

### PR DESCRIPTION
**Pull request by kashifest**
the osv-scanner workflow currently is not producing any meaningful scan output. Manual check revealed it so perhaps not running properly. Changing the workflow it to use osv scanner's workflow (this is recommended way) instead of having own local workflow seems to have solved the issue. This PR updates the workflow accordingly.

### CODE REVIEW

File: .github/workflows/osv-scanner-scan.yml [hunk 1]
```diff
@@ -14,7 +17,8 @@ permissions: {}
 jobs:
   scan-scheduled:
     permissions:
+      actions: read
       contents: read # to fetch code (actions/checkout)
       security-events: write # for uploading SARIF files
     if: ${{ github.repository == 'metal3-io/ip-address-manager' && github.event_name == 'schedule' }}
-    uses: "./.github/workflows/osv-scanner.yml"
+    uses: "google/osv-scanner-action/.github/workflows/osv-scanner-reusable.yml@v1.8.4"
```
Comment (tuminoid) [inline .github/workflows/osv-scanner-scan.yml:24]: Can we use SHA with tag comment here, as it is external action we pull?

File: .github/workflows/osv-scanner-scan.yml [hunk 2]
```diff
@@ -6,6 +6,9 @@ name: OSV-Scanner Scan
 on:
   schedule:
   - cron: "12 12 * * 1"
+  # testing
+  pull_request:
+    branches: [main]
```
Comment (tuminoid) [inline .github/workflows/osv-scanner-scan.yml:11]: Use list format instead.
```suggestion
    branches:
    - main
```

**Comment by kashifest** [comment]
@tuminoid please dont review it yet, this is WIP

**Comment by tuminoid** [comment]
> @tuminoid please dont review it yet, this is WIP

Sorry, was going thru a bunch of them, missed that note.

**Comment by kashifest** [comment]
> Sorry, was going thru a bunch of them, missed that note.

AH no need to be sorry, I meant I will push changes quite a few times here and also it has some test code currently. So no need to waste your energy yet 😄

**Comment by kashifest** [comment]
@tuminoid PTAL now

**Review by tuminoid — COMMENTED** [review]
I think this should work. We'll get some false positives, but we can workaround that with another patch, as discussed.

/approve

**Comment by tuminoid** [comment]
/override metal3-centos-e2e-integration-test-main  metal3-ubuntu-e2e-integration-test-main 
GH action

**Comment by mquhuy** [comment]
/override metal3-centos-e2e-integration-test-main
/override metal3-ubuntu-e2e-integration-test-main
Not related tests

**Pull request merged**